server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -749,9 +749,10 @@ public static boolean hasGaps(TokenStream stream) throws IOException {
         @Override
         public IndexFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName) {
             if (fielddata == false) {
-                throw new IllegalArgumentException("Fielddata is disabled on text fields by default. Set fielddata=true on [" + name()
-                        + "] in order to load fielddata in memory by uninverting the inverted index. Note that this can however "
-                                + "use significant memory. Alternatively use a keyword field instead.");
+                throw new IllegalArgumentException("Text fields are not optimised for aggregations and/or sorting and therefore these "
```

Comment:
This looks good, I had some minor suggestions to try to keep the error message as precise/ short as possible:

> Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [field] in order to load field data by uninverting the inverted index. Note that this can use significant memory.